### PR TITLE
Max height drawing tool

### PR DESCRIPTION
**Pull request by srallen**
A variant of the rectangle drawing tool that is similar to the planet hunters style tool where the rectangle is the full height of the subject, but has adjustable width. This tool is requested for the Intro2Astro activity 4 where students will be classifying an area on a plot. 

You can see it in action with the intended subjects here (with pan and zoom also enabled):
https://max-height-drawing-tool.pfe-preview.zooniverse.org/projects/srallen086/intro2astro-testing/classify

Currently it is also an experimental feature. My hope is to get this merged in next week and to get the project setup for the class to use on April 28.

### CODE REVIEW

File: app/classifier/drawing-tools/column-rectangle.cjsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+module.exports = React.createClass
+  displayName: 'RectangleTool'
+
+  statics:
+    initCoords: null
+
+    defaultValues: ({x, y}) ->
+      x: x
+      y: 0
```
Comment (brian-c) [inline app/classifier/drawing-tools/column-rectangle.cjsx]: Will `y` ever not be `0`? Why record it?

File: app/classifier/drawing-tools/column-rectangle.cjsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      _inProgress: false
+
+    initValid: (mark) ->
+      mark.width >= MINIMUM_WIDTH
+
+  initCoords: null
+
+  render: ->
+    {x, y, width} = @props.mark
+    allowedWidth = if @props.mark.width < MINIMUM_WIDTH then MINIMUM_WIDTH else width
```
Comment (brian-c) [inline app/classifier/drawing-tools/column-rectangle.cjsx]: This decouples the display of the mark from the values in the annotation. Enforce the minimum in the collected data, not the rendered output.

File: app/classifier/drawing-tools/column-rectangle.cjsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
+React = require 'react'
+DrawingToolRoot = require './root'
+DragHandle = require './drag-handle'
+Draggable = require '../../lib/draggable'
+DeleteButton = require './delete-button'
+
+DELETE_BUTTON_DISTANCE = 9 / 10
```
Comment (brian-c) [inline app/classifier/drawing-tools/column-rectangle.cjsx]: This is unused.

File: app/classifier/drawing-tools/column-rectangle.cjsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  handleLeftDrag: (e, d) ->
+    @props.mark.x += d.x / @props.scale.horizontal if @props.mark.width >= MINIMUM_WIDTH
+    @props.mark.width -= d.x / @props.scale.horizontal 
+    @props.onChange @props.mark if @props.mark.width >= MINIMUM_WIDTH
+
+  handleRightDrag: (e, d) ->
+    @props.mark.width += d.x / @props.scale.horizontal
+    @props.onChange @props.mark 
+
+  normalizeMark: ->
```
Comment (brian-c) [inline app/classifier/drawing-tools/column-rectangle.cjsx]: Assuming you're going of the normal rectangle tool: the normal rectangle renders as a `<polyline>` in order the handle negative dimensions while a drag is in progress. We're rendering a `<rect>` here, so I don't think this make too much sense.

**Comment by brian-c** [comment]
A couple overall things:
- [ ] Seems like there's some horsiness when dragging the handles (although not on initial drag) where the recorded coordinates get out of sync with the mouse.
- [ ] Implement `deleteIfOutOfBounds` on the main Draggable.
- [ ] Add an icon in classifier/tasks/drawing/icons
- [ ] Add tool to the drawing task in page/dev-classifier/mock-data

Or if you need this merged pronto for a demo, say the word. Works mostly fine!

**Comment by srallen** [comment]
Thanks. I'll address what I can today, but will need it merged sometime early next week. There will only be one pilot class at NU using it in a project, so some of this may need to come in a later PR.

**Comment by brian-c** [comment]
Sounds good. The data/render out-of-sync comment is the only really scary one.

**Comment by srallen** [comment]
@brian-c I've removed the unnecessary or unused code and reworked the min-width to be in the handlers. I'm still seeing some odd UX behavior if you drag a handle beyond the min-width either direction and back again, but it the annotated mark width and what is rendered seems in sync now.

Gonna start to take a look at some of the other suggestions now.

**Comment by srallen** [comment]
Got the other points done now. This is ready for review again.

**Comment by brian-c** [comment]
Solid, thanks!

**Pull request merged**